packages/host/app/components/operator-mode/container.gts

```diff
@@ -153,6 +168,113 @@ export default class OperatorModeContainer extends Component<Signature> {
+          return [...indexCards, index];
+        }
+        return indexCards;
+      },
+      [] as number[],
+    );
+
+    switch (indexCardIndicies.length) {
+      case 0:
+        // at least one of the top most cards needs to be an index card
+        return;
+
+      case 1:
+        // if only one of the top most cards are index cards, and the index card
+        // has no selections, then the copy state reflects the copy of the top most
+        // card to the index card
+        if (this.selectedCards[indexCardIndicies[0]].length) {
+          // the index card should be the destination card--if it has any
+          // selections then don't show the copy button
+          return;
+        }
+        let destinationItem = topMostStackItems[
+          indexCardIndicies[0]
+        ] as CardStackItem; // the index card is never a contained card
+        let sourceItem =
+          topMostStackItems[indexCardIndicies[0] === LEFT ? RIGHT : LEFT];
+        if (sourceItem.type === 'contained') {
+          return;
+        }
+        return {
+          direction: indexCardIndicies[0] === LEFT ? 'left' : 'right',
+          sources: [sourceItem.card],
+          destinationItem,
+          sourceItem,
+        };
+
+      case 2: {
+        // if both the top most cards are index cards, then we need to analyze
+        // the selected cards from both stacks in order to determine copy button state
+        let sourceStack: number | undefined;
+        for (let [index, stackSelections] of this.selectedCards.entries()) {
+          // both stacks have selections--in this case don't show a copy button
+          if (stackSelections.length > 0 && sourceStack != null) {
+            return;
+          }
+          if (stackSelections.length > 0) {
+            sourceStack = index;
+          }
+        }
+        // no stacks have a selection
+        if (sourceStack == null) {
+          return;
+        }
+        let sourceItem =
+          sourceStack === LEFT
+            ? (topMostStackItems[LEFT] as CardStackItem)
+            : (topMostStackItems[RIGHT] as CardStackItem); // the index card is never a contained card
+        let destinationItem =
+          sourceStack === LEFT
+            ? (topMostStackItems[RIGHT] as CardStackItem)
+            : (topMostStackItems[LEFT] as CardStackItem); // the index card is never a contained card
+
+        // the source and destination are the same
+        if (sourceItem.card.id === destinationItem.card.id) {
+          return;
+        }
+
+        return {
+          direction: sourceStack === LEFT ? 'right' : 'left',
+          sources: this.selectedCards[sourceStack],
+          sourceItem,
+          destinationItem,
+        };
+      }
+      default:
+        throw new Error(
+          `Don't know how to handle copy state for ${this.operatorModeStateService.state.stacks.length} stacks`,
+        );
+    }
+  }
```

Comment:
This code seems fragile to future development, but I think it is fine for now since the future of the dual-stack copy button UI is hazy.

Reply:
Yeah it is tailored to having exactly 2 stacks and the designation being an index card. But given our current requirements, I just don’t know how you’d generalize this.

Reply:
The tests though are incredibly exhaustive of all the states that permit (or do not permit) copying. So I think we’ll have pretty good coverage going forward.